code/modules/reagents/reagent_containers/food/snacks.dm

```diff
@@ -545,7 +545,7 @@
 
 /obj/item/weapon/reagent_containers/food/snacks/chocolatebar/wrapped/valentine/New()
 	..()
-	if(time2text(world.realtime, "MM/DD") != "02/14")
+	if(IS_VALENTINES)
```

Comment:
should be if(!IS_VALENTINES) as it's checking if the date is not valentines day.